aspnetcore/tutorials/min-web-api.md

```diff
@@ -0,0 +1,341 @@
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+* Select **File** > **New Solution**.
+
+`  ![macOS New solution](first-web-api-mac/_static/sln.png)`
+
+* In Visual Studio for Mac earlier than version 8.6, select **.NET Core** > **App** > **API** > **Next**. In version 8.6 or later, select **Web and Console** > **App** > **API** > **Next**.
+
+`  ![macOS API template selection](first-web-api-mac/_static/api_template.png)`
+
+* In the **Configure the new ASP.NET Core Web API** dialog, select the latest .NET Core 5.x **Target Framework**. Select **Next**.
+
+* Enter *TodoApi* for the **Project Name** and then select **Create**.
+
+ ` ![config dialog](first-web-api-mac/_static/2.png)`
+
+[!INCLUDE[](~/includes/mac-terminal-access.md)]
+
+---
+
+### Examine the code
+
+The *Program.cs* file contains the following code:
+
+[!code-csharp[](min-web-api/samples/6.x/Program.cs)]
+
+The following code create a `WebApplicationBuilder` with preconfigured defaults, and builds the web app:
+
+  ```csharp
+  var builder = WebApplication.CreateBuilder(args);
+  var app = builder.Build();
+  ```
+
+`app.MapGet("/", () => "Hello World!");`  maps the `/` endpoint to a method that returns `Hello World!`.
+
+`app.Run();` runs the app.
+
+The preceding code is a complete web app with one endpoint.
+
+### Run the app
+
+# [Visual Studio](#tab/visual-studio)
+
+Press Ctrl+F5 to run without the debugger.
+
+[!INCLUDE[](~/includes/trustCertVS.md)]
+
+  Visual Studio launches:
+
+* The IIS Express web server.
+* The default browser and navigates to `https://localhost:<port>/swagger/index.html`, where `<port>` is a randomly chosen port number.
+
+In a browser, navigate to `/`, for example, `https://localhost:5001/`. `Hello World!` is returned.
+
+# [Visual Studio Code](#tab/visual-studio-code)
+
+[!INCLUDE[](~/includes/trustCertVSC.md)]
+
+Press Ctrl+F5 to run the app. In a browser, go to following URL: [https://localhost:5001/swagger](https://localhost:5001/swagger)
+
+# [Visual Studio for Mac](#tab/visual-studio-mac)
+
+Select **Run** > **Start Debugging** to launch the app. Visual Studio for Mac launches a browser and navigates to `https://localhost:<port>`, where `<port>` is a randomly chosen port number. 
+
+---
+
+`Hello World!` is displayed in the browser.
+
+## Add NuGet packages
+
+NuGet packages must be added to support the database and diagnostics used in this tutorial.
+
+# [Visual Studio](#tab/visual-studio)
+
+* From the **Tools** menu, select **NuGet Package Manager > Manage NuGet Packages for Solution**.
+* Select the **Browse** tab, and then enter `Microsoft.EntityFrameworkCore.InMemory` in the search box. Verify **Incluee prerelease** is checked.
+* Select `Microsoft.EntityFrameworkCore.InMemory` in the left pane.
+* Select the **Project** checkbox in the right pane and then select **Install**.
+* Follow the preceding instructions to add the following packages:
+  * `Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore`
+  * `MinimalValidation`
```

Comment:
Not sure we want to show this in the tutorial? Do we @LadyNaggaga  @DamianEdwards? Maybe we should show fluent validation instead?

Reply:
These aren't in the box yet so, let's remove them. @DamianEdwards  am I wrong ?

Reply:
As @davidfowl says we could decide to show how to use FluentValidation or I could do some work on MinimalValidation and release that as a 1.0 and we could use that. Or we don't talk about validation at all in the tutorial as it's not something we provide in the box today. We know people will ask about it but we also know putting things in the tutorial is in ways an implicit blessing/statement of what we recommend. Likely requires more thought/discussion.